core/promise/consumer.go

```diff
@@ -36,17 +66,49 @@ func (c *Consumer) NewRequest() (requestPtr interface{}) {
 
 // Consume handles requests from endpoint and replies with response
 func (c *Consumer) Consume(requestPtr interface{}) (response interface{}, err error) {
-	// request := requestPtr.(*Request)
+	request, ok := requestPtr.(*Request)
+	if !ok {
+		return nil, errUnsupportedRequest
```

Comment:
Standardised errors is nice, but it's just for logging AFAIK. What about response to sender?
Did you look how responses are standardised at `session/create_dto.go`?

Reply:
Actually not good that we write a lot of logic in dialog transport layer.
I've just refactored `session/create_consumer.go` to be thin and `consumer.SessionManager.Create()` is fat.
Maybe here some `type ValidatePromise func(signedPromise) error`?